services/grafana/client.go

```diff
@@ -0,0 +1,348 @@
+// ctx is used only for cancelation.
+func (c *Client) do(ctx context.Context, method, path string, headers http.Header, body []byte, respBody interface{}) error {
+	u := url.URL{
+		Scheme: "http",
+		Host:   c.addr,
+		Path:   path,
+	}
+	req, err := http.NewRequest(method, u.String(), bytes.NewReader(body))
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	if len(body) != 0 {
+		req.Header.Set("Content-Type", "application/json; charset=utf-8")
+	}
+	for k := range headers {
+		req.Header.Set(k, headers.Get(k))
+	}
+
+	req = req.WithContext(ctx)
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	defer resp.Body.Close() //nolint:errcheck
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	if resp.StatusCode != 200 {
+		return &apiError{
+			code: resp.StatusCode,
+			body: string(b),
+		}
+	}
+
+	if respBody != nil {
+		if err = json.Unmarshal(b, respBody); err != nil {
+			return errors.WithStack(err)
+		}
+	}
+	return nil
+}
+
+// role defines Grafana user role within the organization.
+// Role with more permissions has larger numerical value: viewer < editor, etc.
+type role int
+
+const (
+	none role = iota
+	viewer
+	editor
+	admin
+	grafanaAdmin
+)
+
+func (r role) String() string {
+	switch r {
+	case none:
+		return "None"
+	case viewer:
+		return "Viewer"
+	case editor:
+		return "Editor"
+	case admin:
+		return "Admin"
+	case grafanaAdmin:
+		return "GrafanaAdmin"
+	default:
+		return fmt.Sprintf("unexpected role %d", int(r))
+	}
+}
+
+// getRole returns grafanaAdmin if currently authenticated user is a Grafana (super) admin.
+// Otherwise, it returns a role in the default organization (with ID 1).
+// ctx is used only for cancelation.
+func (c *Client) getRole(ctx context.Context, authHeaders http.Header) (role, error) {
+	// https://grafana.com/docs/http_api/user/#actual-user - works with any authentication
+	var m map[string]interface{}
+	if err := c.do(ctx, "GET", "/api/user", authHeaders, nil, &m); err == nil {
+		if a, _ := m["isGrafanaAdmin"].(bool); a {
+			return grafanaAdmin, nil
+		}
+	}
+
+	// https://grafana.com/docs/http_api/user/#organizations-of-the-actual-user - works with any authentication
+	var s []interface{}
+	if err := c.do(ctx, "GET", "/api/user/orgs", authHeaders, nil, &s); err != nil {
+		return none, err
+	}
+
+	for _, el := range s {
+		m, _ := el.(map[string]interface{})
+		if m == nil {
+			continue
+		}
+
+		// check only default organization (with ID 1)
+		if id, _ := m["orgId"].(float64); id == 1 {
+			role, _ := m["role"].(string)
+			switch strings.ToLower(role) {
+			case "viewer":
+				return viewer, nil
+			case "editor":
+				return editor, nil
+			case "admin":
+				return admin, nil
+			default:
+				return none, nil
+			}
+		}
+	}
+
+	return none, nil
+}
+
+func (c *Client) testCreateUser(ctx context.Context, login string, role role, authHeaders http.Header) (int, error) {
```

Comment:
I'm not sure that test methods should be here, but let it be

Reply:
There is a high chance we will use it very shortly – to create a separate user for pmm-managed to add API keys or separate users for pmm-agents.

Reply:
That is, after renaming and changing.